AnimatedComponents/src/components/floating-button/circular-floating-button/CircularFloatingButton.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import { Dimensions, Image, Pressable, View } from 'react-native'
+import Animated, {
+  Extrapolation,
+  interpolate,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated'
+
+import { PlusIcon } from '@assets'
+import { FloatingButtonProps } from '@types'
+
+import styles from './circularFloatingButton-styles'
+
+const CircularFloatingButton = (props: FloatingButtonProps) => {
+  const {
+    animationTransitionDuration = 500,
+    circleStyle = styles.expandingCircle,
+    contentContainerStyle = styles.contentContainer,
+    iconContainerStyle = styles.iconContainer,
+    iconStyle = styles.icon,
+    icons,
+    isLeft = false,
+  } = props
+
+  const isOpen = useSharedValue(false)
+  const progress = useDerivedValue(() =>
+    isOpen.value
+      ? withTiming(1, { duration: animationTransitionDuration })
+      : withTiming(0, { duration: animationTransitionDuration }),
+  )
+  const { width } = Dimensions.get('window')
+  const FAB_SIZE = 54
```

Comment:
1. Give relevant/clear names and not sort form names.
2. Edit names for animations as per what they are animating. check all animation names
3. create const for animation duration if required.